angr/state_plugins/heap/heap_brk.py

```diff
@@ -76,9 +76,10 @@ def _calloc(self, sim_nmemb, sim_size):
         else:
             size = self.state.solver.eval(sim_size)
 
-        final_size = size * nmemb
-        if final_size > plugin.max_variable_size:
+        if self.state.solver.symbolic(sim_nmemb) or self.state.solver.symbolic(sim_size):
```

Comment:
You should not remove the `final_size > plugin.max_variable_size` check. Otherwise it will always allocate `plugin.max_variable_size` if size is symbolic.

Reply:
Okay, I have added it back.